### PR TITLE
Add @__DIR_P__, @__FILE_P__ and @LOCAL macro

### PR DESCRIPTION
**Pull request by davidanthoff**
CC @vtjnash 

There are some open questions before we can merge this:
- I'm not happy with the names ``@__DIR_P__`` and ``@__FILE_P__``, does anyone have a better idea?
- What do we want to return if this is run from the REPL or ``julia -e``? Right now it always returns ``Path()``, but I'm not sure that is a good idea. Should we instead return ``nothing``?

### CODE REVIEW

**Comment by rofinn** [comment]
I'd opt for just leaving them unexported for now and keep the same names as base.

**Comment by vtjnash** [comment]
I'm not sure why you're pinging me, since I don't currently use this package or currently have any needs that are addressed by it.

**Comment by davidanthoff** [comment]
@vtjnash I'm pinging you because you added ``@LOCAL`` to Electron.jl, where I will remove it soon once this PR here is merged. No need to review anything, I just wanted to let you know where you can find a macro similar to what you had added going forward.

@rofinn I'll go with your new name recommendations, I like them! I do think exporting by default makes sense, after all the whole point of this is to have some short things.

**Comment by davidanthoff** [comment]
> Now that I think about it having just ``@__PATH__``/``@__CWD__`` and ``@__FILEPATH__`` might be best as ``@__FILEPATH__`` could just default to the current file if one isn't specified?

Alright, I have stared at this for a minute, and I just don't understand what this sentence means :) Can you elaborate a bit what you are suggesting?

File: src/path.jl [hunk 1]
```diff
@@ -27,6 +27,38 @@ home() = Path(homedir())
...
 
+"""
+    @__PATH__ -> AbstractPath
+
+@__PATH__ expands to a path with the directory part of the absolute path
+of the file containing the macro. Returns an empty Path if run from a REPL or
+if evaluated by julia -e <expr>.
+"""
+macro __PATH__()
+    :(Path(@__DIR__()===nothing?Path():@__DIR__))
```
[the remark this answers is not in the document]
Comment (davidanthoff) [reply src/path.jl]: Done

**Comment by rofinn** [comment]
Sorry, I was just suggesting that we have `filespec` be an optional argument to the `@__FILEPATH__` macro, rather than having a separate `@LOCAL` macro just for that behaviour.

**Comment by davidanthoff** [comment]
Ah :) Hm, I almost feel having three separate things is cleaner? I'm not too keen on the name ``@LOCAL``, but I think keeping ``@__FILEPATH__`` and ``@__PATH__`` really simple and in line with what is in base would be easier for folks to grok? I think if ``@__FILEPATH__`` did both things it would feel a bit too magical for my tastes?

I do wonder whether there is a better name for ``@LOCAL``, though...

Or, here is another thought: if we used some operator for path concatenating (say ``*``), maybe we could just do without ``@LOCAL`` altogether? Say we had:
````julia
function Base.:*(a::AbstractPath, b::AbstractPath)
    return join(a,b)
end

function Base.:*(a::AbstractPath, b::AbstractString)
    return join(a,Path(b))
end
````
Then one could just write
````julia
@__PATH__() * "foo"
````
Which looks quite natural to me?

**Comment by rofinn** [comment]
I could see introducing `/` as a path concatenation character.

**Comment by davidanthoff** [comment]
How about I remove ``@LOCAL`` for now, so that we can merge the simple things now and iterate a bit more on the other stuff?

**Comment by rofinn** [comment]
Sure, would you be okay with not exporting `@LOCAL` for now? That way we don't need go through a proper deprecation cycle.

**Pull request merged**